benchexec/tools/sv_benchmarks_util.py

```diff
@@ -40,3 +42,132 @@ def get_data_model_from_task(task, param_dict):
                     f"Unsupported data_model '{data_model}' defined for task '{task}'"
                 )
     return None
+
+
+def _partition_input_files(input_files, task_options):
+    """
+    This function partitions the input files into witness and other files.
+    The distinction is based on the file name of the witness file, which
+    is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Tuple of witness files and other files
+    """
+    witness_files = []
+    other_files = []
+    for file in input_files:
+        if Path(file).name == task_options.options.get("witness"):
+            witness_files.append(file)
+        else:
+            other_files.append(file)
+    return witness_files, other_files
+
+
+def get_witness_input_files(task):
+    """
+    This function returns the witness input files from the task.
+    They are identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: List of witness files
+    """
+    witness_files, _ = _partition_input_files(
+        task.input_files_or_identifier, task.options
+    )
+    return witness_files
+
+
+def get_unique_witness(task):
+    """
+    This function returns the unique witness file from the task.
+    It raises an exception if there are multiple witness files.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Unique witness file
+    """
+    witness_files = get_witness_input_files(task)
+    if len(witness_files) > 1:
+        raise UnsupportedFeatureException(
+            "Tool does not support tasks with more than one witness file"
+        )
+    return witness_files[0]
+
+
+def get_non_witness_input_files_or_identifier(task):
+    """
+    This function returns the non-witness input files or the identifier from the task.
+    They consist of all elements which do not match the witness file name.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: List of non-witness files
+    """
+    _, other_files = _partition_input_files(
+        task.input_files_or_identifier, task.options
+    )
+    return other_files
+
+
+def get_non_witness_input_files(task):
+    """
+    This function returns the non-witness input files from the task.
+    They consist of all files which do not match the witness file name.
+    The witness file is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: List of non-witness files
+    """
+    _, other_files = _partition_input_files(task.input_files, task.options)
```

Comment:
I think here we should throw if there is no input file, similarly to how `task.input_files` throws if it would be empty.